packages/neon/neon/lib/src/widgets/user_avatar.dart

```diff
@@ -0,0 +1,199 @@
+            child: NeonCachedApiImage(
+              account: widget.account,
+              cacheKey: 'avatar-$effectiveUsername-${isDark ? 'dark' : 'light'}$size',
+              download: () async {
+                if (isDark) {
+                  return widget.account.client.core.getDarkAvatar(
+                    userId: effectiveUsername,
+                    size: size,
+                  );
+                } else {
+                  return widget.account.client.core.getAvatar(
+                    userId: effectiveUsername,
+                    size: size,
+                  );
+                }
+              },
+            ),
+          ),
+        ),
+        CircleAvatar(
+          radius: widget.size / 2,
+          child: ClipOval(
+            child: NeonCachedApiImage(
+              account: widget.account,
+              cacheKey: 'avatar-$effectiveUsername-${isDark ? 'dark' : 'light'}$size',
+              download: () async {
+                if (isDark) {
+                  return widget.account.client.core.getDarkAvatar(
+                    userId: effectiveUsername,
+                    size: size,
+                  );
+                } else {
+                  return widget.account.client.core.getAvatar(
+                    userId: effectiveUsername,
+                    size: size,
+                  );
+                }
+              },
+            ),
+          ),
+        ),
+        if (widget.showStatus) ...[
+          if (widget.username != null) ...[
+            ResultBuilder<NextcloudUserStatusPublicStatus?>(
+              stream: _userStatusBloc.otherUserStatuses.map(
+                (final statuses) => statuses[effectiveUsername] ?? Result.loading(),
+              ),
+              builder: (final context, final userStatus) => _userStatusIcon(
+                loading: userStatus.loading,
+                error: userStatus.error,
+                userStatus: userStatus.data != null
+                    ? _UserStatus(
+                        icon: userStatus.data!.icon,
+                        status: userStatus.data!.status,
+                      )
+                    : null,
+              ),
+            ),
+          ] else ...[
+            ResultBuilder<NextcloudUserStatusStatus?>(
+              stream: _userStatusBloc.userStatus,
+              builder: (final context, final userStatus) => _userStatusIcon(
+                loading: userStatus.loading,
+                error: userStatus.error,
+                userStatus: userStatus.data != null
+                    ? _UserStatus(
+                        icon: userStatus.data!.icon,
+                        status: userStatus.data!.status,
+                      )
+                    : null,
+              ),
+            ),
+          ],
+        ],
+      ],
+    );
+  }
+
+  Widget _userStatusIcon({
+    required final bool loading,
+    required final Object? error,
+    required final _UserStatus? userStatus,
+  }) {
```

Comment:
Can we rename this to `_userStatusIconBuilder` and change it's interface to the one  `ResultBuilder.builder` expects? Just taking the userStaus and a context.

all the other stuff can be done in the method body. This way we can replace this with a tearoff and I'm sure this would improve code readability quite a bit (also removes the above duplication that should be gone anyways.